include/onnxruntime/core/common/common.h

```diff
@@ -95,6 +99,57 @@ void LogRuntimeError(uint32_t session_id, const common::Status& status, const ch
 #define ORT_WHERE_WITH_STACK \
   ::onnxruntime::CodeLocation(__FILE__, __LINE__, __PRETTY_FUNCTION__, ::onnxruntime::GetStackTrace())
 
+#ifdef ORT_NO_EXCEPTIONS
+
+#define ORT_TRY
+
+// Throw an exception with optional message.
+// NOTE: The arguments get streamed into a string via ostringstream::operator<<
+// DO NOT use a printf format string, as that will not work as you expect.
+#define ORT_THROW(...)                                                                       \
+  do {                                                                                       \
+    std::cerr << ::onnxruntime::OnnxRuntimeException(ORT_WHERE_WITH_STACK,                   \
+                                                     ::onnxruntime::MakeString(__VA_ARGS__)) \
+                     .what()                                                                 \
+              << std::endl;                                                                  \
```

Comment:
Maybe add a TODO in the comment saying we need to consider using something like LogRuntimeError instead of std::cerr when exceptions are disabled so we note that potential change.